crates/amaru/src/bin/amaru/cmd/daemon.rs

```diff
@@ -70,9 +70,15 @@ pub async fn run(
 
     let mut clients: Vec<(String, Arc<Mutex<PeerClient>>)> = vec![];
     for peer in &config.upstream_peers {
-        let client =
-            PeerClient::connect(peer.clone(), config.network.to_network_magic() as u64).await?;
-        clients.push((peer.clone(), Arc::new(Mutex::new(client))));
+        match PeerClient::connect(peer.clone(), config.network.to_network_magic() as u64).await {
+            Ok(client) => {
+                clients.push((peer.clone(), Arc::new(Mutex::new(client))));
+            }
+            Err(e) => {
+                tracing::error!("failed to connect to peer {}: {}", peer, e);
```

Comment:
Tracing is not logging, the philosophy is slightly different.
Besides we must ensure data logged will not [break JSON generation](https://github.com/pragma-org/amaru/issues/161) so tracing full errors is not advised.

Reply:
Hi @jeluard,

I just checked by injecting several _problematic_ characters in the error and they're all escaped correctly when I activate `--with-json-trace`. I did not observe any JSON generation breakage.

Seemingly, one can just inject whatever in this error message and it's escaped so seems to be safe in my tests. Do you have any specific exemple that could help me observe JSON generation breakage? That would help a lot.

Thanks.

Reply:
> How is that escaped? Unless it changed in recent versions it's not. " breaks json.

No idea. I just tested by injecting some problematic characters in the error message and I observe a valid JSON. For instance if I change my line to do this:

```
            .inspect_err(|e| tracing::error!("failed to connect \"to peer {}: {}", peer, "This is a \"problematic\" message with } and '"))?;
```

Running amaru outputs valid JSON:
```
#> cargo run --release -- --with-json-traces daemon  \
  --peer-address=$AMARU_PEER_ADDRESS \
  --network=preprod
{"timestamp":"2025-06-25T15:26:17.805240Z","level":"ERROR","fields":{"message":"failed to connect \"to peer 3.248.56.78:3001: This is a \"problematic\" message with } and '"},"target":"amaru::cmd::daemon"}
[...]
```